File: backend/check_users.py
import sys
import os

# Add the current directory to sys.path
sys.path.append(os.path.dirname(os.path.abspath(__file__)))

from app import create_app
from models import User

def list_users():
    app = create_app()
    with app.app_context():
        users = User.objects()
        if not users:
            print("\nNo users found in the database.")
            return

        print(f"\n{'ID':<30} | {'Name':<20} | {'Email':<30} | {'Role':<10}")
        print("-" * 95)
        for u in users:
            print(f"{str(u.id):<30} | {u.name:<20} | {u.email:<30} | {u.role:<10}")
        print("-" * 95)
        print(f"Total users: {len(users)}\n")

if __name__ == "__main__":
    list_users()

File: backend/src/scripts/checkUsers.js
/**
 * scripts/checkUsers.js
 * CLI script to list all users in the database.
 * Usage: node src/scripts/checkUsers.js
 */

import dotenv from 'dotenv';
import connectDB from '../config/db.js';
import { User } from '../models/index.js';
import mongoose from 'mongoose';

dotenv.config();

async function listUsers() {
  await connectDB();
  const users = await User.find();
  if (!users.length) {
    console.log('\nNo users found in the database.');
    await mongoose.disconnect();
    return;
  }

  console.log(`\n${'ID'.padEnd(30)} | ${'Name'.padEnd(20)} | ${'Email'.padEnd(30)} | ${'Role'.padEnd(10)}`);
  console.log('-'.repeat(95));
  for (const u of users) {
    console.log(`${u._id.toString().padEnd(30)} | ${u.name.padEnd(20)} | ${u.email.padEnd(30)} | ${u.role.padEnd(10)}`);
  }
  console.log('-'.repeat(95));
  console.log(`Total users: ${users.length}\n`);

  await mongoose.disconnect();
}

listUsers().catch((err) => {
  console.error('List users error:', err);
  process.exit(1);
});
